src/presenter.rs

```diff
@@ -136,6 +158,10 @@ impl<'a> Presenter<'a> {
                     CommandSideEffect::None => (),
                 };
             }
+            if let Some(SpeakerNoteChannel::Notifier(notifier)) = self.speaker_notes_channel.as_mut() {
+                let current_slide_idx = self.state.presentation().current_slide_index();
+                notifier.notify_with_custom_event_id(EventId::new(current_slide_idx + 1)).unwrap();
```

Comment:
Did you consider using the [interprocess](https://docs.rs/interprocess/latest/interprocess/index.html) crate instead? I need to read iceoryx2's docs more but this API feels a bit strange. You're communicating via event ids, which is enough for what we're doing but it feels a bit off. (I admit I haven't looked at the docs enough to understand what events mean here).

Reply:
No hard reason to keep using the `EventId` - `iceoryx2` also supports publish/subscribe messaging pattern with different payloads. The examples in the readme/examples dir of the repo show publishing a `usize`. We could easily use this pattern instead.

Reply:
I had come across `interprocess`, but initially chose `iceoryx2` since it appeared to be more popular.
I have since tried the `interprocess` crate and had a quick go at using it in this branch but have not gotten that to work fully yet. Even running the local socket example from `interprocess` I noticed a few things: the "client" must strictly be started after the "server" otherwise it panics, and you have to manually remove the socket it creates between invocations, otherwise it panics.
So `iceoryx2` seems to be more polished in this regard, and I think has a nicer interface.

Reply:
Fair, `interprocess` would also require dealing with all the clients which is pretty annoying.